presto-sqlserver/src/main/java/io/prestosql/plugin/sqlserver/SqlServerClient.java

```diff
@@ -387,4 +420,27 @@ public void setNull(PreparedStatement statement, int index)
             }
         };
     }
+
+    private static class DataCompressionDao
+    {
+        private final Handle handle;
+
+        private DataCompressionDao(Handle handle)
+        {
+            this.handle = requireNonNull(handle, "handle is null");
+        }
+
+        Optional<DataCompression> getDataCompression(JdbcTableHandle table)
+        {
+            return handle.createQuery("" +
+                    "SELECT data_compression_desc FROM sys.partitions p INNER JOIN sys.tables t ON p.object_id = t.object_id " +
+                    "WHERE SCHEMA_NAME(t.schema_id) = :schema AND t.name = :table_name " +
+                    "AND p.partition_number = 1")
```

Comment:
what about other partitions? What they have different `data_compression_desc`? I would do `SELECT DISTINCT data_compression_desc` and in case there is 0 or more than 1 row I would return `Optional.empty()`